projects/storefrontlib/src/cms-components/checkout/guards/checkout-auth.guard.ts

```diff
@@ -21,25 +26,39 @@ export class CheckoutAuthGuard implements CanActivate {
     protected authService: AuthService,
     protected authRedirectService: AuthRedirectService,
     protected checkoutConfigService: CheckoutConfigService,
-    protected activeCartService: ActiveCartService
+    protected activeCartService: ActiveCartService,
+    protected userService: UserService,
+    protected globalMessageService: GlobalMessageService
   ) {}
 
   canActivate(): Observable<boolean> {
     return combineLatest([
       this.authService.getUserToken(),
       this.activeCartService.getAssignedUser(),
+      this.userService.get(),
     ]).pipe(
-      map(([token, user]: [UserToken, User]) => {
+      map(([token, cartUser, user]: [UserToken, User, User | B2BUser]) => {
         if (!token.access_token) {
           if (this.activeCartService.isGuestCart()) {
-            return Boolean(user);
+            return Boolean(cartUser);
           }
           if (this.checkoutConfigService.isGuestCheckout()) {
             this.routingService.go({ cxRoute: 'login' }, { forced: true });
           } else {
             this.routingService.go({ cxRoute: 'login' });
           }
           this.authRedirectService.reportAuthGuard();
+        } else if ('roles' in user) {
+          const roles = (<B2BUser>user).roles;
+          if (roles.indexOf(B2BUserGroup.B2B_CUSTOMER_GROUP) > -1) {
```

Comment:
We could use `includes` instead of `inidexof`.  No big deal.  Your call.

Reply:
ok